### ISSUE TITLE
Multiple external emote integrations

### ISSUE THREAD

**Issue by FFY00** [bimbo-garden/greenhouse#10]
Our main target right now is Twitch, but we should support other websites like Youtube, etc.

#6 proposes a Discord integration, and one of the features would be managing the Discord emotes. Discord here could be just another external emote target, undifferentiated from Twitch for eg.

Features:
- Different integrations could be linked/grouped together, for emote syncing
- Integration-specific emotes (meaning, you could add an emote just to a specific integration, even if it is linked/grouped together with other one)

Questions:
- How do we link the integrations?
  - We should probably have a concept of integration groups. Right?

---

By "integrations" here I mean like Twitch, Youtube, Discord, etc.

**Comment by IZeeLikeEyeZee** [comment]
> How do we link the integrations? We should probably have a concept of integration groups. Right?

Yeah we let people have sets of emotes and assign sets to different integrations as they wish. For something like discord where it won't use our extension for the emotes we'll have to figure out a way to trim down the emote sets if they have more emotes than said server's limit.

We would probably want to create another set dedicated to discord that has the same limit as the server it's connected to, and have an option to link it to a set and have different ways of automatically trimming down the set for users who don't want to select specific emotes for their discord server. Some ideas for that could be prioritization by tag, do whatever (default option, just adds emotes as it can from first to last added), or by prioritization for certain emote artists (to prioritize channel-specific emotes).

We can figure that out later down the line tho tbh, just anything that works would be fine for most people, and I think our main focus should be twitch atm.

**Comment by FFY00** [comment]
Hum, yes, this makes much more sense than the grouping the integrations idea.
